core/src/main/java/org/apache/calcite/prepare/CalcitePrepareImpl.java

```diff
@@ -943,7 +943,8 @@ private int getTypeOrdinal(RelDataType type) {
   }
 
   private static String getClassName(RelDataType type) {
-    return null;
+    // AvaticaParameter requires className field to be not null, see CALCITE-2613
+    return Object.class.getName();
```

Comment:
Is there a method on RelDataType which we could use already to determine the appropriate `Class`? If not, do you think it makes sense to introduce that to the `RelDataType` interface?

Reply:
There is no mapping from SQL types to Java types now. Having it (maybe on RelDataType level, maybe simple static mapping somewhere close to SqlTypeName) could be a good thing but I think that Calcite use cases for that narrows only to satisfy exposing through avatica.  Although this propably will change and then some can think how to implement the mapping.